common/rpc/interceptor/telemetry_test.go

```diff
@@ -85,6 +100,77 @@ func TestEmitActionMetric(t *testing.T) {
 			methodName: metrics.MatchingClientAddWorkflowTaskScope,
 			fullName:   api.WorkflowServicePrefix + queryWorkflow,
 		},
+		{
+			methodName: "UpdateWorkflowExecution",
+			fullName:   api.WorkflowServicePrefix + queryWorkflow,
+		}, {
+			methodName: metrics.HistoryRespondWorkflowTaskCompletedScope,
+			fullName:   api.WorkflowServicePrefix + "RespondWorkflowTaskCompleted",
+			req: &workflowservice.RespondWorkflowTaskCompletedRequest{
+				Commands: []*command.Command{
+					{
+						CommandType: enums.COMMAND_TYPE_PROTOCOL_MESSAGE,
+						Attributes: &command.Command_ProtocolMessageCommandAttributes{
+							ProtocolMessageCommandAttributes: &command.ProtocolMessageCommandAttributes{
+								MessageId: "MESSAGE_ID",
+							},
+						},
+					},
+				},
+				Messages: []*protocolpb.Message{
+					{
+						Id:   "MESSAGE_ID",
+						Body: &updateAcceptanceMessageBody,
+					},
+				},
+			},
+			expectEmitMetrics: true,
+		},
+		{
+			methodName: metrics.HistoryRespondWorkflowTaskCompletedScope,
+			fullName:   api.WorkflowServicePrefix + "RespondWorkflowTaskCompleted",
+			req: &workflowservice.RespondWorkflowTaskCompletedRequest{
+				Commands: []*command.Command{
+					{
+						CommandType: enums.COMMAND_TYPE_PROTOCOL_MESSAGE,
+						Attributes: &command.Command_ProtocolMessageCommandAttributes{
+							ProtocolMessageCommandAttributes: &command.ProtocolMessageCommandAttributes{
+								MessageId: "MESSAGE_ID",
+							},
+						},
+					},
+				},
+				Messages: []*protocolpb.Message{
+					{
+						Id:   "MESSAGE_ID",
+						Body: &updateRejectionMessageBody,
+					},
+				},
+			},
+			expectEmitMetrics: true,
+		},
+		{
+			methodName: metrics.HistoryRespondWorkflowTaskCompletedScope,
+			fullName:   api.WorkflowServicePrefix + "RespondWorkflowTaskCompleted",
+			req: &workflowservice.RespondWorkflowTaskCompletedRequest{
+				Commands: []*command.Command{
+					{
+						CommandType: enums.COMMAND_TYPE_PROTOCOL_MESSAGE,
+						Attributes: &command.Command_ProtocolMessageCommandAttributes{
+							ProtocolMessageCommandAttributes: &command.ProtocolMessageCommandAttributes{
+								MessageId: "MESSAGE_ID",
+							},
+						},
+					},
+				},
+				Messages: []*protocolpb.Message{
+					{
+						Id:   "MESSAGE_ID",
+						Body: &updateCompletionMessageBody,
+					},
+				},
+			},
```

Comment:
Update completion is not charged.

Reply:
I would move this comment to the code by adding `case *updatepb.Response` to the `select` above.